mobile/lib/common/status_icon_button.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter/material.dart';
+import 'package:get_10101/common/channel_status_notifier.dart';
+import 'package:get_10101/common/service_status_notifier.dart';
+import 'package:get_10101/common/status_screen.dart';
+import 'package:get_10101/bridge_generated/bridge_definitions.dart' as rust;
+import 'package:provider/provider.dart';
+
+class StatusIconButton extends StatelessWidget {
+  const StatusIconButton({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final channelStatusNotifier = context.watch<ChannelStatusNotifier>();
+    final serviceStatusNotifier = context.watch<ServiceStatusNotifier>();
+
+    final overallStatus = serviceStatusNotifier.overall();
+
+    return IconButton(
+      icon: channelStatusNotifier.isClosing() || overallStatus == rust.ServiceStatus.Offline
+          ? const Icon(Icons.thermostat, color: Colors.red)
+          : overallStatus == rust.ServiceStatus.Unknown
+              ? const Icon(Icons.thermostat, color: Colors.yellow)
+              : const Icon(Icons.thermostat),
```

Comment:
This might be cleaner as an if/else or switch - this is a bit nested which makes it harder to read

Reply:
because of the indentation I find it quite clear, but of course it might be subjective. I'll keep it in mind for the future.